niceman/distributions/redhat.py

```diff
@@ -0,0 +1,339 @@
+        """
+
+        # TODO: Expand the listing of 3rd party repos as popular ones in the
+        # neuroscience community become apparent.
+        third_party_repos = {
+            'epel/x86_64': 'epel-release'
+        }
+
+        # Get a list of the enabled repos on the system.
+        out, _ = session.execute_command(['yum', 'repolist'])
+        system_repos = []
+        skip = True
+        for line in out.splitlines():
+            if line.startswith('repo id'):
+                skip = False
+                continue
+            if skip:
+                continue
+            system_repos.append(line.split()[0])
+
+        # Try to install any repos listed in the spec that are missing on the system.
+        for source in self.sources:
+            if source.id not in system_repos:
+                if source.id in third_party_repos:
+                    session.execute_command(['yum', 'install', '-y',
+                        third_party_repos[source.id]])
+                else:
+                    lgr.error("Unable to install source repo \"%s\" found in \
+spec but not in resource environment.", source.id)
+
+    def install_packages(self, session, use_version=True):
+        """
+        Install the packages associated to this distribution by the provenance
+        into the environment.
+
+        Parameters
+        ----------
+        session : object
+        use_version : bool, optional
+          Use version information if provided.
+          TODO: support outside or deprecate
+            
+        """
+        package_specs = []
+
+        for package in self.packages:
+            package_spec = package.pkgid
+            package_specs.append(package_spec)
+
+        # Doing in one shot to fail early if any of the versioned specs
+        # couldn't be satisfied
+        lgr.debug("Installing %s", ', '.join(package_specs))
+        session.execute_command(
+            # TODO: Pull env out of provenance for this command.
+            ['yum', 'install', '-y'] + package_specs,
+            # env={'DEBIAN_FRONTEND': 'noninteractive'}
+        )
+
+    def satisfies_package(self, package):
+        """return True if this distribution (self) satisfies the requirements 
+        of the passed package"""
+        if not isinstance(package, Package):
+            raise TypeError('satisfies_package() requires a package argument')
+        if not isinstance(package, RPMPackage):
+            return False
+        return any([p.satisfies(package) for p in self.packages])
+
+    def satisfies(self, other):
+        """return True if this distribution (self) satisfies the requirements 
+        of the other distribution (other)"""
+        if not isinstance(other, Distribution):
+            raise TypeError('satisfies() requires a distribution argument')
+        if not isinstance(other, RedhatDistribution):
+            return False
+        return all(map(self.satisfies_package, other.packages))
```

Comment:
eh... that is what I still hope #144 could address generically without requiring each distribution to provide redundant logic